### PR TITLE
[VL] Enable map_zip_with, zip_with functions

### PR DESCRIPTION
**Pull request by zhli1142015**
## What changes were proposed in this pull request?

Enable map_zip_with, zip_with functions.

## How was this patch tested?

UT.

### CODE REVIEW

**Comment by zhli1142015** [comment]
```
- map_zip_with function - map of primitive types *** FAILED ***
  Results do not match for query:
  Timezone: sun.util.calendar.ZoneInfo[id="America/Los_Angeles",offset=-28800000,dstSavings=3600000,useDaylight=true,transitions=185,lastRule=java.util.SimpleTimeZone[id=America/Los_Angeles,offset=-28800000,dstSavings=3600000,useDaylight=true,startYear=0,startMode=3,startMonth=2,startDay=8,startDayOfWeek=1,startTime=7200000,startTimeMode=0,endMode=3,endMonth=10,endDay=1,endDayOfWeek=1,endTime=7200000,endTimeMode=0]]
  Timezone Env: 
  
  == Parsed Logical Plan ==
  'Project [unresolvedalias('map_zip_with('m1, 'm2, lambdafunction((lambda 'k = (lambda 'v1 + lambda 'v2)), lambda 'k, lambda 'v1, lambda 'v2, false)), Some(org.apache.spark.sql.Column$$Lambda$7458/1239312138@6b4e6bc7))]
  +- Project [_1#139294 AS m1#139299, _2#139295 AS m2#139300]
     +- LocalRelation [_1#139294, _2#139295]
  
  == Analyzed Logical Plan ==
  map_zip_with(m1, m2, lambdafunction((namedlambdavariable() = (namedlambdavariable() + namedlambdavariable())), namedlambdavariable(), namedlambdavariable(), namedlambdavariable())): map<int,boolean>
  Project [map_zip_with(m1#139299, m2#139300, lambdafunction((cast(lambda k#139303 as bigint) = (lambda v1#139304L + cast(lambda v2#139305 as bigint))), lambda k#139303, lambda v1#139304L, lambda v2#139305, false)) AS map_zip_with(m1, m2, lambdafunction((namedlambdavariable() = (namedlambdavariable() + namedlambdavariable())), namedlambdavariable(), namedlambdavariable(), namedlambdavariable()))#139306]
  +- Project [_1#139294 AS m1#139299, _2#139295 AS m2#139300]
     +- LocalRelation [_1#139294, _2#139295]
  
  == Optimized Logical Plan ==
  Project [map_zip_with(_1#139294, _2#139295, lambdafunction((cast(lambda k#139303 as bigint) = (lambda v1#139304L + cast(lambda v2#139305 as bigint))), lambda k#139303, lambda v1#139304L, lambda v2#139305, false)) AS map_zip_with(m1, m2, lambdafunction((namedlambdavariable() = (namedlambdavariable() + namedlambdavariable())), namedlambdavariable(), namedlambdavariable(), namedlambdavariable()))#139306]
  +- LocalRelation [_1#139294, _2#139295]
  
  == Physical Plan ==
  VeloxColumnarToRowExec
  +- ^(2823) ProjectExecTransformer [map_zip_with(_1#139294, _2#139295, lambdafunction((cast(lambda k#139303 as bigint) = (lambda v1#139304L + cast(lambda v2#139305 as bigint))), lambda k#139303, lambda v1#139304L, lambda v2#139305, false)) AS map_zip_with(m1, m2, lambdafunction((namedlambdavariable() = (namedlambdavariable() + namedlambdavariable())), namedlambdavariable(), namedlambdavariable(), namedlambdavariable()))#139306]
     +- ^(2823) InputIteratorTransformer[_1#139294, _2#139295]
        +- ^(2823) InputAdapter
           +- ^(2823) RowToVeloxColumnar
              +- ^(2823) LocalTableScan [_1#139294, _2#139295]
  
  == Results ==
  
  == Results ==
  !== Correct Answer - 4 ==                   == Gluten Answer - 4 ==
  !struct<>                                   struct<map_zip_with(m1, m2, lambdafunction((namedlambdavariable() = (namedlambdavariable() + namedlambdavariable())), namedlambdavariable(), namedlambdavariable(), namedlambdavariable())):map<int,boolean>>
  ![Map(10 -> null, 8 -> false, 4 -> null)]   [Map(3 -> false, 6 -> true, 8 -> true)]
  ![Map(5 -> null)]                           [Map(4 -> null, 8 -> false, 10 -> null)]
  ![Map(8 -> true, 3 -> false, 6 -> true)]    [Map(5 -> null)]
   [null]                                     [null] (GlutenSQLTestsTrait.scala:97)
```

**Comment by zhli1142015** [comment]
cc @PHILO-HE  and @rui-mo , thanks.

File: gluten-ut/spark32/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -278,6 +278,8 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude("shuffle function - array for primitive type not containing null")
     .exclude("shuffle function - array for primitive type containing null")
     .exclude("shuffle function - array for non-primitive type")
+    // Velox sorts rows by key for primitive data types, which disrupts the original row sequence.
+    .exclude("map_zip_with function - map of primitive types")
```
Comment (rui-mo) [inline gluten-ut/spark32/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:282]: Do we need to rewrite this test? Thanks.

**Review by PHILO-HE — APPROVED** [review]
Looks good!

File: gluten-ut/spark32/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala (hunk 1 above)
Comment (zhli1142015) [reply gluten-ut/spark32/src/test/scala/org/apache/gluten/utils/velox/VeloxTestSettings.scala:282]: Added, thanks.

**Pull request merged**